### PR TITLE
add Bukkit worldmigrator docs

### LINKED ISSUE #431: Document World Migrator
Currently, the docs state that CraftBukkit/Spigot worlds must be manually migrated, by copying subfolders to specific locations. However, as of Beta, SpongeForge and SpongeVanilla container a migrator which can automatically convert worlds to the proper format.

### PR DESCRIPTION
**Pull request by Tzky**
This PR resolves #431. 

This adds documentation of the automated World Migrator to import Bukkit worlds into SpongeVanilla or SpongeForge.

I haven't used this feature yet, so any input is welcome, especially if the instructions are correct.

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/c57d231e/index.html.

Here are some links to the pages that were modified:

- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/c57d231e/server/getting-started/migrating.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/432#issuecomment-170685250), not to the direct url to the preview._

**Comment by Zidane** [comment]
I wrote the Migrator...so I guess I should be the one to say :p.

1. Migrator will look into the ```bukkit.yml``` for a config key called ```world-container``` and will search that folder for worlds to transfer. If that file isn't around (or something caused it to fail to read), **then I will use the root folder of the server** (which is CraftBukkit standard).

2. Perform migration. At this point, I move over worlds from what I call the ```world container``` into the folder defined in server.properties as ```level-name```.

3. Bukkit put things in weird places (with weird names). I must perform two important fixes. Keep in mind this is based on assumptions (due to Bukkit's structure).
    * The first fix is to rename any folder whose name starts with ```level-name``` property and ends with a Vanilla dimension name (```_nether```/```_the_end```) to ```DIM-1``` and ```DIM1``` respectively.
    * The next fix is to migrate up the ```region``` data within a Bukkit Vanilla nether/the_end. Bukkit puts this data into ```DIM-1\region``` and ```DIM1\region``` respectively whereas Vanilla/Forge expect ```region``` to be in the **root** of the world's folder structure.

4. The last step is to generate a config file for the new world.

I want to stress that **Sponge does not load the migrated world**. This is due to the fact that Bukkit stored little to none information on the World itself. What generator did it use? What dimension did it display to the client? It is up to a worlds plugin such as Project Worlds from the forums to properly load the migrated world.

File: source/server/getting-started/migrating.rst [hunk 1]
```diff
@@ -11,6 +11,9 @@ to Sponge.
    :depth: 2
    :local:
 
+.. warning::
+  Please update your server files **before** migrating. In case something goes wrong, you still have your backups!
```
Comment (Zidane) [inline source/server/getting-started/migrating.rst]: s/update/backup

File: source/server/getting-started/migrating.rst [hunk 2]
```diff
@@ -29,19 +32,43 @@ Worlds
...
+   these fixes are based on *assumptions* (due to Bukkit's structure).
+
+   * The first fix is to rename any folder whose name starts with ``level-name`` property and ends with a Vanilla
+     dimension name (``_nether``/``_the_end``) to ``DIM-1`` and ``DIM1`` respectively.
+
+   * The second fix is to migrate up the ``region`` data within a Bukkit Vanilla nether/the_end. Bukkit puts this data
+     into ``DIM-1\region`` and ``DIM1\region`` respectively whereas Vanilla/Forge expect ``region`` to be in the
+     **root** of the world's folder structure.
+
+#. The last step is to generate a config file for the new world.
```
Comment (Zidane) [inline source/server/getting-started/migrating.rst]: Actually I am going to stop generating config files. This should only ever happen if another plugin asks to load it.

Comment (Tzky) [inline source/server/getting-started/migrating.rst]: I assume you want a plugin to generate the config then?

File: source/server/getting-started/migrating.rst [hunk 3]
```diff
@@ -29,19 +32,42 @@ Worlds
...
 places the nether (typically ``world_nether``) and the end (typically ``world_the_end``) dimensions within the
-``world`` folder.
+``world`` folder. However Bukkit and Spigot *don't* use this system to save the worlds, thus migration is needed.
+
+SpongeForge and SpongeVanilla provide a fully automated conversion script which converts your worlds for you.
+This is how it works:
+
+1. Shutdown your Bukkit or Spigot server and Backup the entire folder.
+
+#. Place SpongeForge or SpongeVanilla in the folder where you ran your old server, remove the Bukkit or Spigot jars.
```
Comment (Zidane) [inline source/server/getting-started/migrating.rst]: Maybe link to the docs about setting up a SpongeForge/SpongeVanilla server here.

File: source/server/getting-started/migrating.rst [hunk 4]
```diff
@@ -29,19 +32,42 @@ Worlds
...
+
+   * The first fix is to rename any folder whose name starts with ``level-name`` property and ends with a Vanilla
+     dimension name (``_nether``/``_the_end``) to ``DIM-1`` and ``DIM1`` respectively.
+
+   * The second fix is to migrate up the ``region`` data within a Bukkit Vanilla nether/the_end. Bukkit puts this data
+     into ``DIM-1\region`` and ``DIM1\region`` respectively whereas Vanilla/Forge expect ``region`` to be in the
+     **root** of the world's folder structure.
+
+The migrator can't provide all needed configuration values. This is the reason why you need to change several parameters
+by hand to make the world work properly upon loading it on SpongeForge or SpongeVanilla. Usually a
```
Comment (Zidane) [inline source/server/getting-started/migrating.rst]: Usually implies there is a case for when this is not needed....and you are right! Only for what I've determined as Vanilla Nether/End coming from Bukkit (the ones that get renamed).

For any other worlds added via a Bukkit plugin, you must load them via a Worlds plugin on Sponge.

Comment (Tzky) [inline source/server/getting-started/migrating.rst]: Changed wording here, i guess it's best to just encourage users to use a world management plugin. Will hopefully save us headaches in the future.

File: source/server/getting-started/migrating.rst (hunk 3 above)
Comment (Tzky) [inline source/server/getting-started/migrating.rst]: aaand done.

Comment (Inscrutable) [inline source/server/getting-started/migrating.rst]: It is a bit misleading when it says to put SpongeForge in the folder. "Install" might be a better phrase, or include the messy clarification that you need Forge there and SpongeForge in the mods.
Also, typo on next line: "propberly" -> properly.

File: source/server/getting-started/migrating.rst [hunk 5]
```diff
@@ -153,8 +193,8 @@ Migrating to SpongeForge
 ------------------------
 
 You must first ensure you are running a version of Forge that is compatible with the version of Sponge your plan to use.
```
Comment (Inscrutable) [inline source/server/getting-started/migrating.rst]: It may be old, but it's still a typo: "your" -> you. Also, Sponge -> SpongeForge.

**Pull request merged**